Plugin crashed: interface {} is nil

## Description

Latest patch has introduced to a crash situation. Maybe README is incomplete?

## ⚠️ Note

Before you submit an issue, please perform the following first:

1. Remove the local `.terraform` directory (! ONLY if state is stored remotely, which hopefully you are following that best practice!): `rm -rf .terraform/`
2. Re-initialize the project root to pull down modules: `terraform init`
3. Re-attempt your terraform plan or apply and check if the issue still persists

## Versions

- Module version [Required]:

- Terraform version:
1.1.8
- Provider version(s):
provider registry.terraform.io/hashicorp/aws v4.17.1


## Reproduction Code [Required]

<!-- REQUIRED -->

Steps to reproduce the behavior:

1. Previously had been using the `capacity_providers = ["FARGATE", "FARGATE_SPOT"]` and `container_insights = true`.
2. Replaced with `fargate_capacity_providers` attribute per new README
3. `terraform validate` passes
4. `terraform apply` <- crash happens here

## Expected behavior

No crash

## Actual behavior

Empty `{}` object's default value is `nil` and causing a panic

### Terminal Output Screenshot(s)

```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # module.ecs.module.ecs.aws_ecs_cluster.this[0] will be updated in-place
  ~ resource "aws_ecs_cluster" "this" {
        id                 = "arn:aws:ecs:us-east-1:XXXXXXXXXXXXXXX:cluster/dev"
        name               = "dev"
        tags               = {
            "Environment" = "dev"
        }
        # (3 unchanged attributes hidden)

      + configuration {
        }

        # (1 unchanged block hidden)
    }

  # module.ecs.module.ecs.aws_ecs_cluster_capacity_providers.this[0] will be updated in-place
  ~ resource "aws_ecs_cluster_capacity_providers" "this" {
      ~ capacity_providers = [
          - "FARGATE_SPOT",
            # (1 unchanged element hidden)
        ]
        id                 = "dev"
        # (1 unchanged attribute hidden)

      + default_capacity_provider_strategy {
          + base              = 0
          + capacity_provider = "FARGATE"
          + weight            = 100
        }
    }

Plan: 0 to add, 2 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.ecs.module.ecs.aws_ecs_cluster.this[0]: Modifying... [id=arn:aws:ecs:us-east-1:XXXXXXXXXXXXXXX:cluster:cluster/dev]
╷
│ Error: Request cancelled
│ 
│   with module.ecs.module.ecs.aws_ecs_cluster.this[0],
│   on .terraform/modules/ecs.ecs/main.tf line 5, in resource "aws_ecs_cluster" "this":
│    5: resource "aws_ecs_cluster" "this" {
│ 
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵
Releasing state lock. This may take a few moments...

Stack trace from the terraform-provider-aws_v4.17.1_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 40 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/ecs.expandClusterConfiguration({0xc0021d02e0, 0x94ce2e8, 0x2})
	github.com/hashicorp/terraform-provider-aws/internal/service/ecs/cluster.go:557 +0x113
github.com/hashicorp/terraform-provider-aws/internal/service/ecs.resourceClusterUpdate(0xc002ccb500, {0x8027680, 0xc00037f500})
	github.com/hashicorp/terraform-provider-aws/internal/service/ecs/cluster.go:332 +0x2a5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xa5a7760, {0xa5a7760, 0xc002cfbb60}, 0xd, {0x8027680, 0xc00037f500})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/resource.go:729 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000d01880, {0xa5a7760, 0xc002cfbb60}, 0xc001692750, 0xc002ccb380, {0x8027680, 0xc00037f500})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/resource.go:847 +0x9e5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000296930, {0xa5a76b8, 0xc002c97100}, 0xc002c94d20)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/grpc_provider.go:1021 +0xe3c
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc001ae3860, {0xa5a7760, 0xc002cfb380}, 0xc0001b65b0)
	github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/tf5server/server.go:812 +0x56b
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x92321e0, 0xc001ae3860}, {0xa5a7760, 0xc002cfb380}, 0xc002c8d860, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00013e700, {0xa6b4510, 0xc000cac1a0}, 0xc002186c60, 0xc001aef830, 0x1030dda0, 0x0)
	google.golang.org/grpc@v1.45.0/server.go:1282 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc00013e700, {0xa6b4510, 0xc000cac1a0}, 0xc002186c60, 0x0)
	google.golang.org/grpc@v1.45.0/server.go:1619 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.45.0/server.go:921 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.45.0/server.go:919 +0x294

Error: The terraform-provider-aws_v4.17.1_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

```

## Additional context

N/A

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -23,7 +23,7 @@ repos:
           - '--args=--only=terraform_standard_module_structure'
           - '--args=--only=terraform_workspace_remote'
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v4.2.0
+    rev: v4.3.0
     hooks:
       - id: check-merge-conflict
       - id: end-of-file-fixer

--- a/main.tf
+++ b/main.tf
@@ -12,7 +12,7 @@ resource "aws_ecs_cluster" "this" {
 
     content {
       dynamic "execute_command_configuration" {
-        for_each = try([configuration.value.execute_command_configuration], [])
+        for_each = try([configuration.value.execute_command_configuration], [{}])
 
         content {
           kms_key_id = try(execute_command_configuration.value.kms_key_id, null)

